paddle/fluid/operators/lrn_op.cc

```diff
@@ -207,6 +201,24 @@ class LRNOp : public framework::OperatorWithKernel {
         OperatorWithKernel::IndicateVarDataType(ctx, "X"), ctx.GetPlace(),
         layout_, library_);
   }
+
+  framework::OpKernelType GetKernelTypeForVar(
+      const std::string& var_name, const Tensor& tensor,
+      const framework::OpKernelType& expected_kernel_type) const override {
+#ifdef PADDLE_WITH_MKLDNN
+    if ((expected_kernel_type.data_layout_ == framework::DataLayout::kMKLDNN) &&
+        (tensor.layout() != framework::DataLayout::kMKLDNN)) {
+      auto attrs = Attrs();
+      auto ar = paddle::framework::AttrReader(attrs);
+      const std::string data_format = ar.Get<std::string>("data_format");
+      return framework::OpKernelType(
+          expected_kernel_type.data_type_, tensor.place(),
+          framework::StringToDataLayout(data_format));
```

Comment:
Why you need to get tensor layout from "data_format"? Isn't it set in any place based on this attribute?

Reply:
apprently it is not. When "data_format=NHWC" still Tensors that contain data stored according to that order are having layout set to : kNCHW. The thing is that I do not know current reading of Tensor::layout. In MKL-DNN integration we use kNCHW and kMKLDNN . As for kAnyLayout I'm not sure of it purpose now.